### PR TITLE
FINERACT-2576: Add unit test for MoneyTest in fineract-core

### PR DESCRIPTION
**Pull request by paultofunmi**
## Description

This PR adds a comprehensive `MoneyTest` suite covering all major operations on the `Money` class, including parameterized edge cases for floating-point precision.

**Test coverage added:**

- `plus(double)` — parameterized cases including negative values and floating-point precision edge cases 
- `total(Money...)` varargs — aggregation of multiple `Money` instances, and exception thrown on empty input
- `total(Iterable<Money>)` — iterable aggregation including single-element and empty-list exception cases
- `roundToMultiplesOf(double, int)` — parameterized cases plus NaN return for negative values
- `roundToMultiplesOf(BigDecimal, Integer)` — parameterized cases including zero and negative multiples
- `roundToMultiplesOf(Money, int)` — with and without `MathContext`, including zero-multiple passthrough
- `ceiling(double, double)` — NaN cases and valid cases covering zero, exact multiples, and rounding up
- `floor(double, double)` — NaN cases and valid cases covering negative step sizes and floating-point steps
- `getAmountDefaultedToNullIfZero()` — null for zero, non-null with correct value for non-zero
- `dividedBy(BigDecimal/double/long)` — identity optimisation (returns same instance when dividing by 1), correct results for division by 2
- `multipliedBy(BigDecimal/double/long)` — with and without `MathContext`, identity optimisation for multiplier of 1
- `multiplyRetainScale(BigDecimal/double)` — scale preservation at currency decimal places
- `percentageOf(BigDecimal)` — 50%, 100%, and 0% cases
- `negated()` — positive, negative, and zero (returns same instance for zero)
- `abs()` — positive, negative, zero, and with `MathContext`
- `zero()` — instance method with and without `MathContext`
- `getMc()` — non-null `MathContext` assertion
- `copy()` — value equality, same currency, different instance; copy with `BigDecimal` and `double` overrides

All tests are pure unit tests with no Spring context or database dependencies.

Fixes: https://issues.apache.org/jira/browse/FINERACT-2576

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [X] Write the commit message as per [our guidelines](https://github.com/apache/fineract/blob/develop/CONTRIBUTING.md#pull-requests)
- [X] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.
- [X] Create/update [unit or integration tests](https://fineract.apache.org/docs/current/#_testing) for verifying the changes made.
- [X] Follow our [coding conventions](https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions).
- [X] No Swagger/API changes (N/A — pure unit tests)
- [X] [This PR must not be a "code dump"]  — single focused change (https://cwiki.apache.org/confluence/display/FINERACT/Pull+Request+Size+Limit). Large changes can be made in a branch, with assistance. Ask for help on the [developer mailing list](https://fineract.apache.org/#contribute).
- [x] Commits are GPG signed
Your assigned reviewer(s) will follow our [guidelines for code reviews](https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide).

### CODE REVIEW

**Review by Aman-Mittal — APPROVED** [review]
LGTM, only unit tests are added.

File: fineract-core/src/test/java/org/apache/fineract/organisation/monetary/domain/MoneyTest.java [hunk 1]
```diff
@@ -109,4 +121,379 @@ void testAddMoney() {
...
+    @Test
+    void testMultiplyRetainScaleDouble() {
+        Money result = tenDollars.multiplyRetainScale(1.5, MATH_CONTEXT);
+        assertEquals(2, result.getAmount().scale());
+        assertEquals(0, result.getAmount().compareTo(new BigDecimal("15.00")));
+    }
+
+    @Test
+    void testMultiplyRetainScaleDoubleOne() {
+        // BigDecimal.valueOf(1.0) == BigDecimal.ONE, so returns same instance
```
Comment (budaidev) [inline fineract-core/src/test/java/org/apache/fineract/organisation/monetary/domain/MoneyTest.java]: this comment is not exactly true, since these 2 values are a little bit different. For this scenario it works, because in the calculation uses compareTo. I would suggest to remove the comment.

File: fineract-core/src/test/java/org/apache/fineract/organisation/monetary/domain/MoneyTest.java [hunk 2]
```diff
@@ -109,4 +121,379 @@ void testAddMoney() {
         Money result = tenDollars.add(oneDollar, MATH_CONTEXT);
         assertEquals(0, result.getAmount().compareTo(new BigDecimal("11.00")), "Should correctly add two Money amounts");
     }
+
+    @ParameterizedTest(name = "plus(double {0}) = {1}")
+    @CsvSource({ "0.0,   10.00", "1.0,   11.00", "0.01,  10.01", "5.5,   15.50", "10.0,  20.00", "0.99,  10.99", "100.0, 110.00",
```
Comment (budaidev) [inline fineract-core/src/test/java/org/apache/fineract/organisation/monetary/domain/MoneyTest.java]: Can we do some kind of formatting here, which does it in pairs?
Something like:
```
 // @formatter:off                                                                                                                                                                                                
  @CsvSource({                                                                                                                                                                                                     
      "0.0,   10.00",                                                                                                                                                                                                                                                                                                                                                                                    
      "0.01,  10.01",                                                                                                                                                                                              
      ...     
  })                                                                                                                                                                                                               
  // @formatter:on  
```

Comment (paultofunmi) [reply fineract-core/src/test/java/org/apache/fineract/organisation/monetary/domain/MoneyTest.java]: I initially put each input on a new line but ./gradlew spotlessApply applies predefined formatting removing whitespaces and lines

Comment (budaidev) [reply fineract-core/src/test/java/org/apache/fineract/organisation/monetary/domain/MoneyTest.java]: Maybe with // @formatter:off  and @formatter:on

Comment (paultofunmi) [reply fineract-core/src/test/java/org/apache/fineract/organisation/monetary/domain/MoneyTest.java]: This is done now @budaidev

**Review by budaidev — APPROVED** [review]

**Comment by budaidev** [comment]
LGTM

**Comment by Aman-Mittal** [comment]
@adamsaghy can you look at this PR. Me and attilia have given our reviews.
